### PR TITLE
add kamigakushi

### PR DESCRIPTION
**Pull request by alabahuy**
## Description

Add kamigakushi PCB with processor RP2040 and support via

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kamigakushi/info.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    },
+    "processor": "RP2040",
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0765",
+        "vid": "0x414C"
+    },
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (zvecr) [inline keyboards/kamigakushi/info.json:52]: From PR checklist
> * use of LAYOUT_all is only valid when providing additional layout macros
>     * providing only LAYOUT_all is invalid - especially when implementing the additional layouts within 3rd party tooling

File: keyboards/kamigakushi/rules.mk [hunk 2]
```diff
@@ -0,0 +1,4 @@
+SERIAL_DRIVER = vendor
+WS2812_DRIVER = vendor
+OLED_ENABLE = no
```
Comment (zvecr) [inline keyboards/kamigakushi/rules.mk:3]:
```suggestion
```

File: keyboards/kamigakushi/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2018-2023 QMK (@qmk)
```
Comment (zvecr) [inline keyboards/kamigakushi/keymaps/default/keymap.c:1]:
```suggestion
// Copyright 2023 Alabahuy (@Alabahuy)
```

File: keyboards/kamigakushi/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2018-2023 QMK (@qmk)
```
Comment (zvecr) [inline keyboards/kamigakushi/keymaps/via/keymap.c:1]:
```suggestion
// Copyright 2023 Alabahuy (@Alabahuy)
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/kamigakushi/info.json (hunk 1 above)
Comment (alabahuy) [reply keyboards/kamigakushi/info.json:52]: can i use LAYOUT_65_ansi_7u for this layout ? @zvecr

Comment (zvecr) [reply keyboards/kamigakushi/info.json]: Referencing <https://github.com/qmk/qmk_firmware/blob/master/layouts/default/readme.md>,


`LAYOUT_65_ansi_blocker_tsangan` matches better?

Comment (alabahuy) [reply keyboards/kamigakushi/info.json]: right, thanks :)

File: keyboards/kamigakushi/keymaps/via/keymap.c (hunk 4 above)
Comment (alabahuy) [reply keyboards/kamigakushi/keymaps/via/keymap.c]: Done :) @zvecr

**Comment by alabahuy** [comment]
any update @zvecr ?

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by alabahuy** [comment]
Thanks @drashna @fauxpark :)

**Pull request merged**